### DISCUSSION TITLE
derivative of seq with respect to ddp

### DISCUSSION THREAD

**Discussion by GhiathMonnet** [thelfer/tfel#573] [General]
Hi All,
I am using @Brick StandardElasticity in my implicit implementation and calculate the equivalent stress seq in a standard way.
For the calculation of the Jacobian, I need to compute dfp_ddp which involves the derivative of seq with respect to dp : dseq_ddp.
Since in the french document "behavior.pdf" eq. 17 we have : 
seq = sqrt(6 * (B|B)) - 3 mu theta dp,
the derivative : ( dseq_ddp = - 3 * mu * theta ) is correct ?
Many thanks.
Ghiath

**Comment by thelfer** [comment]
Equation 17 results from th reduction of the implicit system to a scalar equation, by eliminating the elastic strain. This equation is used 

The brick StandardElasticity does assume that the elastic strain is the first integration variable, so it has not been eliminated. It also assumes the stress is computed by the Hook law as a function of the elastic strain only. Thus:

~~~~
dseq_mts_ddeel = dseq_dsig_mts * dsig_ddeel_mts * deel_mts_ddeel
~~~~

where mts meants middle of time step, i.e. at `t+theta*dt`, `eel_mts = eel + theta * deel` .

This leads to:

~~~~
dseq_mts_ddeel = n_mts * D_mts * theta
~~~~

where n_mts is the von Mison normal and D_mts is the elastic stiffness. For an isotropic material, this can be simplified further:

~~~~
dseq_mts_ddeel = theta* mu_mts * n_mts 
~~~~

where mu_mts is the shear modulus.

**Reply by GhiathMonnet** [reply]
Thanks Thomas. I think I understood already what you said. My question was about dseq_mts_ddp. 
However, I did not know how in StandardElasticity seq_mts was implicitly calculated : 

1) seq_mts  = mu * sqrt(6 * (B|B)) - 3 mu theta dp    (in this case dseq_ddp = 3mu*theta)
or as : 
2) seq_mts  = D* (eel + theta *deel)   (in this case dseq_ddp = 0.0)

If I understand correctly, your are saying that in @Brick StandardElasticity : seq_mts  = D* (eel + theta *deel). 
In this case I conclude that dseq_mts_ddp = 0. Is it right ?

**Reply by thelfer** [reply]
Ok, my answer was not clear enough. Formula 1 is used by some DSLs in MFront, `IsotropicMisesCreep` for instance. We could use it in other DSLs, such as the `Default DSL` or the `ImplicitII` DSL.

However, if you use the StandardElasticity brick, you must use Formula 2. In this case, `dseq_mts_ddp = 0`

**Reply by GhiathMonnet** [reply]
Great !! It is clear for me. 
Many thanks.
Ghiath 👍
